includes/options/class-amp-reader-theme-rest-controller.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * Reader theme controller.
+ *
+ * @package AMP
+ * @since 1.6.0
+ */
+
+/**
+ * AMP reader theme REST controller.
+ *
+ * @since 1.6.0
+ */
+final class AMP_Reader_Theme_REST_Controller extends WP_REST_Controller {
+
+	/**
+	 * Reader themes provider class.
+	 *
+	 * @var AMP_Reader_Themes
+	 */
+	private $reader_themes;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param AMP_Reader_Themes $reader_themes AMP_Reader_Themes instance to provide theme data.
+	 */
+	public function __construct( $reader_themes ) {
+		$this->reader_themes = $reader_themes;
+		$this->namespace     = 'amp-wp/v1';
+		$this->rest_base     = 'reader-themes';
+	}
+
+	/**
+	 * Sets up hooks.
+	 */
+	public function init() {
+		add_filter( 'amp_reader_themes', [ $this, 'prepare_default_reader_themes_for_rest' ] );
+	}
+
+	/**
+	 * Overrides data for default themes.
+	 *
+	 * @param array $themes Default reader themes.
+	 * @return array Filtered default reader themes.
+	 */
+	public function prepare_default_reader_themes_for_rest( $themes ) {
+		return array_map( [ $this, 'prepare_default_reader_theme_for_rest' ], $themes );
+	}
+
+	/**
+	 * Overrides data for a default theme.
+	 *
+	 * @param array $theme Reader theme data.
+	 * @return array Filtered reader theme data.
+	 */
+	public function prepare_default_reader_theme_for_rest( $theme ) {
```

Comment:
This is tricky. I suppose these are needed because the description is too long?

Could we instead just re-use what is in core but truncate it with an ellipsis (even with CSS) and allow them to expand to read all? This will be less to maintain and easier for sites to extend with their own existing themes.

Reply:
Yes, this did feel like a temporary solution at best. I'll try the ellipsis thing.

Reply:
This worked out with `wp_trim_words`. The only problem is that now the hardcoded theme data in AMP_Reader_Themes is not translatable. Do you think we should go ahead and use `themes_api` instead of hardcoding?

Reply:
Yes, let's use the data returned by `themes_api`. 

As for the ellipsis, that's also tricky.  If the text is limited to one line we could use `text-overflow:ellipsis`?